### DISCUSSION TITLE
Camera Record status in YAML - behaviour changed

### DISCUSSION THREAD

**Discussion by jonandel** [blakeblackshear/frigate#6833] [Ask A Question]
HI All,

Long time Frigate used (2+ years).  Four cameras. 
Recent update pushed seems to change behaviour of global vs the camera settings.

My problem:
Frigate UI show a red icon for 'recorging' and tells me that I cant turn on recording as its 'Recording must be enabled in the Config to be turned on'.
However, it is as Yaml sets global recording = true for all cameras.
But, two of the cameras I want to be recording= off, so in the camera elements, this is set. (see below)

I do this, as by default and at startup, or system restart etc I want two cameras to always record, and the other two only when explicitly turned on via the UI, or in my Automations...for example, when the HA Manual Alarm is set.  I have automations that turn the two cameras to record on vs off depending if in Alarm armed state.

Strangely, the behavious for snapshots seems to be as it was in a previous release.
Is this a bug ?

My Yaml has:
.....
record:
  enabled: True
  retain:
    days: 0
  events: 
    retain: 
      default: 16 
      mode: active_objects
.......
cameras:
  study:
    ffmpeg:
      inputs:
        - path: rtsp://admin:mypassword@192.168.2.99:554/h264Preview_01_main
          roles:
            - record
        - path: rtsp://admin:mypassword@192.168.2.99:554/h264Preview_01_sub
          roles:
            - detect
    detect:
      width: 640
      height: 480
    objects:
      track:
        - person
    #By default dont record or keep snapshots (only do when alarm is on)
    record:
      enabled: False
    snapshots:
      enabled: False

**Comment by NickM-27** [comment]
This is working as expected. In previous versions those two cameras would say recording is enabled but they would not actually record anything. Recording must be set to on in the config if you want a camera to be recording. To have a camera start off with recording disabled you'll need to have an automation do that. 

And this doesn't apply to snapshots because snapshots don't require a separate process to be running

**Reply by jonandel** [reply]
Interesting - thanks for replying Nick.
Original testing was about 18 months to 2 years ago - and looking at recent recordings, I cant see if recordings were or were not being taken when the automation set the specific cameras....I would have thought setting the recording=true should spawn a new process.  
But anyway - Ill take your word for it.

Any suggestion on what even to trigger the 'set those specific cameras to NOT record' in HA ?  HA startup (surely too early) ? Frigate startup ? Automation on the specific detection (and then turn off if not in Alarm set mode) seems resource intensive ?
thanks again in advance of any help....

**Reply by NickM-27** [reply]
> I would have thought setting the recording=true should spawn a new process.

it does not do this, which is exactly why it must be enabled in the config to be able to be toggled later

> Any suggestion on what even to trigger the 'set those specific cameras to NOT record' in HA ? HA startup (surely too early) ? Frigate startup ? 

I use the state of the recording switch going from unavailable to true in HA automation
